aws-rds-cfn-common/src/main/java/software/amazon/rds/common/logging/RequestLogger.java

```diff
@@ -3,110 +3,130 @@
+        }
+    }
+
     public void logAndThrow(Throwable throwable) {
         log(throwable);
         ExceptionUtils.rethrow(throwable);
     }
 
-    public <RequestT, ResponseT, ClientT> BiFunction<RequestT, ProxyClient<ClientT>, ResponseT> log(
-            final BiFunction<RequestT, ProxyClient<ClientT>, ResponseT> makeServiceCallMethod) {
-        return (request, proxyClient) -> {
-            ResponseT result = null;
-            try {
-                log("RdsRequest", request, ImmutableMap.of("Operation", request.getClass().getSimpleName()));
-                result = makeServiceCallMethod.apply(request, proxyClient);
-                log("RdsResponse", result, ImmutableMap.of("Operation", result.getClass().getSimpleName()));
-            } catch (Exception e) {
-                logAndThrow(e);
-            }
-            return result;
-        };
-    }
-
-    public void log(Throwable throwable) {
-        try {
-            JSONObject jsonObject = new JSONObject(jsonPrinter.print(throwable));
-            jsonObject.put(REQUEST_DATA_MARKER, requestData);
-            logMessage(jsonObject.toString());
-        } catch (Throwable caughtThrowable) {
+    public void log(Object object) {
+        try{
+            DEFAULT_LOG_RULE_SET.handle(object);
+        } catch (Throwable throwable) {
             logMessage(throwable);
         }
     }
 
-    public void log(String message) {
-        log(MESSAGE_MARKER, message);
+    public void log(String event, Object object) {
+        log(event, object, null);
     }
 
-    public void log(String marker, Object object) {
-        log(marker, object, null);
-    }
-
-    public void log(String marker, Object object, Map<String, String> additionalFields) {
+    public void log(String event, Object object, Map<String, String> additionalFields) {
         try {
             String objectAsString = jsonPrinter.print(ObjectUtils.defaultIfNull(object, StringUtils.EMPTY));
-            final JSONObject jsonLog = new JSONObject();
-            jsonLog.put(marker, new JSONObject(objectAsString));
-            Optional.ofNullable(additionalFields).orElse(Collections.emptyMap())
-                    .entrySet().stream().forEach(entry -> jsonLog.put(entry.getKey(), entry.getValue()));
-            jsonLog.put(REQUEST_DATA_MARKER, requestData);
+            final JSONObject jsonLog = new JSONObject(objectAsString);
+            jsonLog.put(EVENT, event);
+            addMapToJson(jsonLog, additionalFields);
+            addMapToJson(jsonLog, requestDataMap);
             logMessage(jsonLog.toString());
         } catch (Throwable throwable) {
             logMessage(throwable);
         }
     }
 
+    private void addMapToJson(final JSONObject jsonObject, final Map<String, String> map) {
+        Optional.ofNullable(map).orElse(Collections.emptyMap())
+                .entrySet().stream().forEach(entry -> jsonObject.put(entry.getKey(), entry.getValue()));
+    }
+
```

Comment:
Yes there is a good one line code
`objectMapper.readValue(jsonObject, LinkedHashMap.class)`
But it will require `@SuppressWarnings("unchecked")` as well. Should I apply it?